src/agent/rustjail/src/container.rs

```diff
@@ -1074,24 +1052,24 @@ fn do_exec(args: &[String]) -> ! {
         .collect();
     let a: Vec<&CStr> = sa.iter().map(|s| s.as_c_str()).collect();
 
-    if let Err(e) = unistd::execvp(p.as_c_str(), a.as_slice()) {
-        match e {
+    unistd::execvp(p.as_c_str(), a.as_slice())
+        .map_err(|e| match e {
             nix::Error::Sys(errno) => {
                 std::process::exit(errno as i32);
             }
             _ => std::process::exit(-2),
-        }
-    }
+        })
+        .ok();
 
     unreachable!()
 }
 
 fn update_namespaces(logger: &Logger, spec: &mut Spec, init_pid: RawFd) -> Result<()> {
     info!(logger, "updating namespaces");
-    let linux = match spec.linux.as_mut() {
-        None => return Err(anyhow!("Spec didn't contain linux field")),
-        Some(l) => l,
-    };
+    let linux = spec
+        .linux
+        .as_mut()
+        .ok_or(anyhow!("Spec didn't contain linux field"))?;
```

Comment:
And please use `ok_or_else` instead of `ok_or`, because: 

> Arguments passed to `ok_or` are **eagerly evaluated**; if you are passing the result of a function call, it is recommended to use `ok_or_else`, which is lazily evaluated.

Reply:
done